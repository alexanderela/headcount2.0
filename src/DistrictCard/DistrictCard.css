.DistrictCard {
	display: flex;
	align-items: center;
	flex-direction: column;
	justify-content: center;
	font-family: 'Open Sans', sans-serif;
	text-align: center;
	width: 15rem;
	height: 26rem;
	margin: 0 auto 1.5rem;
	box-shadow: 0px 1px 1px 2px #8d8686;
	border-radius: 10px;
}

.card-location {
	text-align: center;
	overflow-wrap: break-word;
	margin: 1rem 0.5rem 0.75rem;
}

.card-stats {
	margin: 0 auto 1.5rem;

}

.greater-than-point-5 {
	color: green;
	margin: 0.25rem auto;
}

.less-than-point-5 {
	color: red;
	margin: 0.25rem auto;

}

