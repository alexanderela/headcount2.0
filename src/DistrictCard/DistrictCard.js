import React from 'react';
import './DistrictCard.css';
import PropTypes from 'prop-types';

const DistrictCard = ({ location, stats }) => {
	const statsKeys = Object.keys(stats)
	const schoolData = statsKeys.map((stat) => {
		
		return <p 
			className={(stats[stat] > 0.5) ? 'greater-than-point-5' : 'less-than-point-5'}>
			{stat}: {stats[stat]}</p>
	})
	console.log(schoolData)

	return (
		<div className='DistrictCard'>
			<h2 className='card-location'>{location}</h2>
			{schoolData}
		</div>
	)
}

DistrictCard.proptypes = {
	location: PropTypes.string.isRequired,
	stats: PropTypes.object.isRequired
}

export default DistrictCard;

			// className={(stats[stat] > 0.5) ? '.greater-than-point-5' : '.less-than-point-5'}>
			// {stat}: {stats[stat]}</p>