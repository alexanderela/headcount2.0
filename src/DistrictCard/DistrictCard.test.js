import React from 'react';
import ReactDOM from 'react-dom';
import DistrictCard from './DistrictCard.js';
import { shallow } from 'enzyme';

describe('DistrictCard', () => {
	let wrapper;
	let expectedId;

	beforeEach(() => {
		expectedId = 5
		wrapper = shallow(<DistrictCard key='COLORADO' schoolName='colorado' stats={2004: 0.24, 2005: 0.543, 2006: 0.789}/>)
	})

	it('matches the snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('renders each entry of data', () => {
		expect(wrapper.find(p).length).toEqual(3)
	})

	it.skip('adds a class for data less than 0.5', () => {
		//show instance of data less than 5 with one class
		expect(wrapper.find(p).hasClass('less-than-point-5')).toEqual(true)
	})

	it.skip('adds a class for data greater than 0.5', () => {
		//show instance of data greater than 5 with one class
		expect(wrapper.find('.district-card-stat').hasClass('greater-than-point-5')).toEqual(true)
	})	

})

