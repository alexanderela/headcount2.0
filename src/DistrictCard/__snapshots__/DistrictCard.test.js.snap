// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistrictCard matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DistrictCard
    schoolName="colorado"
    stats={
      Object {
        "2004": 0.24,
        "2005": 0.543,
        "2006": 0.789,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="card-location"
        />,
        Array [
          <p
            className="less-than-point-5"
          >
            2004
            : 
            0.24
          </p>,
          <p
            className="greater-than-point-5"
          >
            2005
            : 
            0.543
          </p>,
          <p
            className="greater-than-point-5"
          >
            2006
            : 
            0.789
          </p>,
        ],
      ],
      "className": "DistrictCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "card-location",
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2004",
            ": ",
            0.24,
          ],
          "className": "less-than-point-5",
        },
        "ref": null,
        "rendered": Array [
          "2004",
          ": ",
          0.24,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2005",
            ": ",
            0.543,
          ],
          "className": "greater-than-point-5",
        },
        "ref": null,
        "rendered": Array [
          "2005",
          ": ",
          0.543,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2006",
            ": ",
            0.789,
          ],
          "className": "greater-than-point-5",
        },
        "ref": null,
        "rendered": Array [
          "2006",
          ": ",
          0.789,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="card-location"
          />,
          Array [
            <p
              className="less-than-point-5"
            >
              2004
              : 
              0.24
            </p>,
            <p
              className="greater-than-point-5"
            >
              2005
              : 
              0.543
            </p>,
            <p
              className="greater-than-point-5"
            >
              2006
              : 
              0.789
            </p>,
          ],
        ],
        "className": "DistrictCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "card-location",
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2004",
              ": ",
              0.24,
            ],
            "className": "less-than-point-5",
          },
          "ref": null,
          "rendered": Array [
            "2004",
            ": ",
            0.24,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2005",
              ": ",
              0.543,
            ],
            "className": "greater-than-point-5",
          },
          "ref": null,
          "rendered": Array [
            "2005",
            ": ",
            0.543,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2006",
              ": ",
              0.789,
            ],
            "className": "greater-than-point-5",
          },
          "ref": null,
          "rendered": Array [
            "2006",
            ": ",
            0.789,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
